packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -1595,6 +1612,7 @@ class SemanticsNode extends AbstractNode with DiagnosticableTreeMixin {
         scrollPosition ??= node._scrollPosition;
         scrollExtentMax ??= node._scrollExtentMax;
         scrollExtentMin ??= node._scrollExtentMin;
+        indexInParent ??= node._indexInParent;
```

Comment:
This seems odd. The descendant may be the third child of its parent, but that doesn't mean that the node this is getting merged into is the third child of its parent?